app/models/conversion_host.rb

```diff
@@ -335,7 +340,12 @@ def ansible_playbook(playbook, extra_vars = {}, miq_task_id = nil, auth_type = '
     extra_vars.each { |k, v| command << " --extra-vars '#{k}=#{v}'" }
 
     result = AwesomeSpawn.run(command)
-    raise unless result.exit_status.zero?
+
+    if result.failure?
+      error_message = result.error.presence || result.output
+      _log.error("#{result.command_line} => #{error_message}")
+      raise
+    end
```

Comment:
This can be replaced with `AwesomeSpawn.run!(command)` if you want it to log an error and raise an exception on failure.

Reply:
Alright, will take a look!

Reply:
@agrare Unfortunately, we can't use it. It assumes that command line apps diligently send errors to stderr, which is a nice thought in theory.

https://github.com/ManageIQ/awesome_spawn/blob/master/lib/awesome_spawn.rb#L104

Unfortunately, the reality is that many command line apps - including `ansible_playbook` - send everything to stdout. I've run into this before over the years, I think even with scvmm and powershell.

Reply:
@djberg96 if it is a common issue we should update AwesomeSpawn to log stdout also then, the idea behind `.run!` is to replace exactly this block from having to be repeated

/cc @carbonin

Reply:
@agrare We can suggest it sure. But for now, this will have to do.

Reply:
Agree that awesome spawn could definitely log more. I'd say this is fine for now though. Probably not worth holding this for a gem release.

To the point about awesome spawn usage though ... you could probably replace a lot of the command building above with the native command building awesome spawn does by default. Additionally `--extra-vars` supports json and yaml, so that might be easier than building a ton of the same option. Also should help with the inevitable quoting issues you'll run into if someone uses a single quote in an extravar ...

Reply:
https://github.com/ManageIQ/awesome_spawn/pull/45